src/Microsoft.Health.Dicom.Core/Features/HealthCheck/BackgroundServiceHealthCheckCache.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Extensions.Caching.Memory;
+
+
+namespace Microsoft.Health.Dicom.Core.Features.HealthCheck
+{
+
+    public class BackgroundServiceHealthCheckCache
```

Comment:
Overall I think this class can be refactored. You don't need to expose "keys" for these methods as the keys are well-defined.  Instead I would probably change the APIs to be like:

```csharp
Task<DateTimeOffset> GetOrAddOldestTimeAsync(Func<Task<DateTimeOffset>> getOldestTime);

Task<int> GetOrAddRetriesAsync(Func<Task<int>> getRetries);
```